### ISSUE TITLE
GTK 3.21 Caja runtime warnings

### ISSUE THREAD

**Issue by lukefromdc** [mate-desktop/caja#561]
These are the current runtime warnings with GTK 3.21 and  Caja, Exact values for underallocations (both required and requested) change with the theme, this is with my UbuntuStudio_Legacy theme. Not a theme issue, Adwaita does this too with larger numbers:

(caja:14357): Gtk-WARNING **: Theme parsing error: <data>:1:51: The style property GtkStatusbar:shadow-type is deprecated and shouldn't be used anymore. It will be removed in a future version

(caja:14357): Gtk-WARNING **: Allocating size to GtkButton 0x21e8500 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(caja:14357): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to underallocate CajaPathBar's child GtkButton 0x21e8500. Allocation is 19x22, but minimum required size is 25x21.

(caja:14357): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to underallocate GtkButton's child GtkArrow 0x2225120. Allocation is 9x16, but minimum required size is 15x15.

(caja:14357): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to underallocate CajaPathBar's child GtkButton 0x21e8500. Allocation is 19x22, but minimum required size is 25x21.

(caja:14357): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to underallocate CajaPathBar's child GtkButton 0x21e8500. Allocation is 19x22, but minimum required size is 25x21.

(caja:14357): Gtk-WARNING **: Allocating size to GtkTable 0x1eec700 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(caja:14357): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to underallocate CajaPathBar's child GtkButton 0x21e8500. Allocation is 19x22, but minimum required size is 25x21.
##

**Comment by monsta** [comment]
Ok... how's it with latest 3.22.x?

**Comment by lukefromdc** [comment]
On running "sudo caja" to force a new instance as root, I get only these three warnngs, which both  Nemo and Nautilus also get:

```
(caja:20474): Gtk-WARNING **: Failed to get the GNOME session proxy: The name org.gnome.SessionManager is not owned

(caja:20474): Gtk-WARNING **: Failed to get the Xfce session proxy: The name org.xfce.SessionManager is not owned

(caja:20474): Gtk-WARNING **: Failed to get an inhibit portal proxy: The name org.freedesktop.portal.Desktop is not owned
```
I also found this in xsession-errors, related to thumbnailing a video:

```
(caja:5642): Gdk-CRITICAL **: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
[avi @ 0x2261ea0] non-interleaved AVI
[swscaler @ 0x2295dc0] deprecated pixel format used, make sure you did set range correctly
[avi @ 0x84eec0] non-interleaved AVI
[swscaler @ 0x87f660] deprecated pixel format used, make sure you did set range correctly
Seeking in video failed Operation not permitted, will use first frame.
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

```

Caja starting with the session did NOT put the warnings a root instance gets into xsession-errors, presumably because it is started by the session manager. Only other text Caja generates in the session is these non-error messages for initializing extensions:
```

Initializing caja-image-converter extension
** Message: Initializing gksu extension...
Initializing caja-open-terminal extension
```

**Comment by shirishag75** [comment]
Hi all, 

I have been following the ticket. On gtk 3.22 the following caja warnings are there -

```
┌─[shirish@debian] - [~] - [6609]
└─[$] cat .xsession-errors | grep caja
                                                                                                
(caja:2524): Gdk-CRITICAL **: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397e79b760' has no handler with id '52276'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b9240a0' has no handler with id '49513'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b90b1a0' has no handler with id '49523'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b8ce8c0' has no handler with id '49533'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b7cc420' has no handler with id '49503'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397f26d4e0' has no handler with id '54421'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397d560ce0' has no handler with id '54431'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397d46de80' has no handler with id '54441'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397bd90420' has no handler with id '54451'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397ba8a930' has no handler with id '66513'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b9240a0' has no handler with id '4254'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b90b1a0' has no handler with id '4264'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b8ce8c0' has no handler with id '4274'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397b7cc420' has no handler with id '4244'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397bd90420' has no handler with id '63111'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397f219c20' has no handler with id '63071'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397d504680' has no handler with id '63081'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397f26d4e0' has no handler with id '63091'
(caja:2524): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x56397d560ce0' has no handler with id '63101'
```

Hope this helps as well.

**Comment by lukefromdc** [comment]
https://github.com/shirishag75, which version of Caja are you running? A whole bunch of those "no handler" warnings were fixed just before 1.16 came out.

**Comment by shirishag75** [comment]
I am on Debian testing/stretch hence caja is at -

```
┌─[shirish@debian] - [~] - [6808]
└─[$] caja --version                                                                                                                  
MATE caja 1.16.1
```

**Comment by lukefromdc** [comment]
This commit missed the 1.16 branch and is not in 1.16.1:
https://github.com/mate-desktop/caja/commit/24bb72b321c044479728fd1d43c941afc0309f18
It stops gsignal warnings on closing a window and probably needs to be cherrypicked for 1.16.2. I am running 1.17.1 and have this commit as a result.

Count the number of gobject/gsignal warnings you have, close a Caja window, count them again. if the number has increased, this is the problem.

**Comment by shirishag75** [comment]
How, because if I try -

```
┌─[shirish@debian] - [~] - [6850]
└─[$] caja                                                                                                                          
┌─[shirish@debian] - [~] - [6851]
```

Nothing happens as can be seen above. Caja doesn't get displayed . It passes it self-tests though I believe -

```
┌─[shirish@debian] - [~] - [6852]
└─[$] caja -c                                                                                                                        
running caja_self_check_file_utilities
running caja_self_check_file_operations
running caja_self_check_directory
running caja_self_check_file
running caja_self_check_icon_container
running caja_self_check_file_utilities
running caja_self_check_file_operations
running caja_self_check_directory
running caja_self_check_file
running caja_self_check_icon_container
```

The only way to use caja is via the Places > some option or via the caja launcher.

**Comment by monsta** [comment]
@lukefromdc: yes, I didn't pick that commit as I wasn't sure if it's safe to include in a point release... Same's about https://github.com/mate-desktop/caja/commit/6e142d59d7408e53ab1e579dba62517b6f59e307.

**Comment by raveit65** [comment]
We should pick every commit which reduce warnings to 1.16 if possible.
Spamming logs is a bug and is annoying for admins who have to look in logs for servers.

**Comment by shirishag75** [comment]
I'm running Debian mate so don't think the `$caja --no-desktop -n` happens unless the Debian-mate developers/maintainers have done that although dunno why it would do that is beyond me.

**Comment by lukefromdc** [comment]
1.16.1 builds with libunique, not with GtkApplication as in 1.17.1 so at least we know the gtkApplication work did not cause this caja exiting on invocation that you are reporting. Still wonder what is doing it though. Do you have icons on the desktop working normally? If so Caja is opening fine from the session.

**Comment by shirishag75** [comment]
I'm running Debian mate so don't think the `$caja --no-desktop -n` happens unless the Debian-mate developers/maintainers have done that although dunno why it would do that is beyond me. 

Now what I did was remove .xsession-errors , rebooted the system and again grepped for caja errors but found nothing. I then used the Places > Home icon so that caja opens. When I closed caja I got the following errors -

```
┌─[shirish@debian] - [~] - [6788]
└─[$] cat .xsession-errors | grep caja                                                                                                
┌─[shirish@debian] - [~] - [6789]
└─[$] cat .xsession-errors | grep caja                                                                                                
┌─[shirish@debian] - [~] - [6789]
└─[$] cat .xsession-errors | grep caja                                                                                                
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566e9d0dd80' has no handler with id '6491'
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566e9ba6a80' has no handler with id '4252'
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566e9b8f940' has no handler with id '4262'
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566e9b8ce80' has no handler with id '4272'
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566e9ba66a0' has no handler with id '4242'
(caja:2604): GLib-GObject-WARNING **: /build/glib2.0-YqrQRU/glib2.0-2.50.2/./gobject/gsignal.c:2641: instance '0x5566ea012f80' has no handler with id '8309'
```

The first one is empty when only the desktop is on, no caja opened. 

The second one grepping caja is when caja is opened through the icon. 

and the last one is whose messages you are seeing.

**Comment by lukefromdc** [comment]
Those are the warnings suppressed by the commits that were not picked for 1.16 and came out after it was branched, that is now confirmed. There were warnings on closing Caja and on closing caja windows and that's them.

**Comment by shirishag75** [comment]
All icons working fine. The issue I reported about caja is still there. Should I file a new report for that ?

**Comment by lukefromdc** [comment]
The last three warnings about not finding GNOME/XFCE session managers etc on a root (or terminal when not previously autostarted) invocation of Caja are stopped by this GTK 3.22 commit, which should appear in GTK 3.22.6:
https://git.gnome.org/browse/gtk+/commit/?h=gtk-3-22&id=30b5187e60a3811c938730764c1caa3ee101b228
GtkApplication: Lack of optional components shouldn't warn
From that point forward, up-to-date Caja with up to date GTK 3.22 should throw no warnings whatsoever on opening a root instance in terminal. These warnings came in with GtkApplication and were shared with Nemo and Nauitilus. Now, five years after GtkApplication was first used in Nautilus these are fixed in GTK3. Si

**Comment by raveit65** [comment]
Imo, this is all fixed with 1.18.0.
If not hit me for closing this......
